pkg/runcfanotify/runcfanotify.go

```diff
@@ -119,6 +119,41 @@ func cmdlineFromPid(pid int) []string {
 	return strings.Split(string(cmdline), "\x00")
 }
 
+// WatchContainerTermination watches a container for termination and generate
+// an event on the notifier. This is useful for containers that were started
+// before the notifier is started.
+func (n *RuncNotifier) WatchContainerTermination(containerID string, containerPID int) {
+	go n.watchContainerTermination(containerID, containerPID)
+}
+
+func (n *RuncNotifier) watchContainerTermination(containerID string, containerPID int) {
+	pidfd, _, errno := unix.Syscall(unix.SYS_PIDFD_OPEN, uintptr(containerPID), 0, 0)
+	if errno != 0 {
+		log.Errorf("pidfd_open on %d returned %v", containerPID, errno)
+		return
+	}
+	defer unix.Close(int(pidfd))
+
+	for {
+		fds := []unix.PollFd{
+			{
+				Fd:      int32(pidfd),
+				Events:  unix.POLLIN,
+				Revents: 0,
+			},
+		}
```

Comment:
The man page says:

> The field revents is an output parameter, filled by the kernel with the events that actually occurred.

But it does not explicitly say that the kernel will initialise `revents` to zero at the beginning of `poll()` so I find it cleaner to initialise `revents` each time.

But in this code, I don't check `revents` since there can be only one possible event... so it does not make any differences.